streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -2734,7 +2734,13 @@ abstract class ZStream[-R, +E, +O](val process: ZManaged[R, Nothing, ZIO[R, Opti
               chunks >>= store
             case Done =>
               Pull.end
-          }
+          }.onInterrupt(
+            ref.get.flatMap {
+              case Previous(fiber) => fiber.interrupt
+              case Current(fiber)  => fiber.interrupt
+              case _               => ZIO.unit
+            }
+          )
```

Comment:
Sorry for this PR dragging out. It's probably better for this finalizer to be attached to the construction of the `Ref`; this way the fiber is interrupted even if the stream is interrupted "in-between" pulls. What do you think?

Reply:
@iravid I think you mean:
```scala
Ref.make[State](NotStarted).toManaged {
  _.get.flatMap {
    case Previous(fiber) => fiber.interrupt
    case Current(fiber)  => fiber.interrupt
    case _               => ZIO.unit
  }
}
```
Don't you?

Reply:
Yes, exactly

Reply:
@iravid makes sense, done. Also added a new test case for the fiber interruption.